web/views.py

```diff
@@ -340,6 +340,12 @@ def user_can_participate(self):
 
     def dispatch(self, request, *args, **kwargs):
 
+        # Give participant 4 days from starting any study before session times out, to allow for cases where the
+        # parent does instructions etc. but then waits until a convenient time to do the actual test with the
+        # child.
+        if not request.user.is_researcher:
+            request.session.set_expiry(60 * 60 * 24 * 4)
```

Comment:
Looks good! Two things:
1) Perhaps would make sense to put this code in [`self.setup`](https://github.com/django/django/blob/master/django/views/generic/base.py#L82) (make sure to `super()`, since `setup` is not really a hook)?
2) Just to double check, this means that every time a participant visits _any_ experiment view, their login expiration gets reset to 4 days in the future. As long as we're cool with that - looks great to me!

Reply:
1. Moved to setup - thanks! 
2. That's correct - I think I'm ok with that but it does mean that someone could in principle stay logged in indefinitely by participating every <4 days. I guess the concern is that an attacker got access to a session but doesn't actually have login credentials, but especially as this is a participant-only extension I don't *think* we'd be making anything substantially more secure by having them log out and back in?